tensorflow/python/keras/losses.py

```diff
@@ -1414,7 +1414,7 @@ def categorical_hinge(y_true, y_pred):
   y_true = math_ops.cast(y_true, y_pred.dtype)
   pos = math_ops.reduce_sum(y_true * y_pred, axis=-1)
   neg = math_ops.reduce_max((1. - y_true) * y_pred, axis=-1)
-  return math_ops.maximum(0., neg - pos + 1.)
+  return math_ops.maximum(neg - pos + 1., 0.)
```

Comment:
`maximum(a, b)` should have the same behaviour as `maximum(b, a)`. I'm not sure this is the right fix.

Reply:
Yes its not behaviour change of maximum. while comparing, 0. is in  float32 and because of mixed precision calculated quantity is in float16, hence the missmatch [op tries to match dtype x to the y and throws error]. I have pointed out this in colab notebook above, you can check.

Reply:
https://colab.research.google.com/drive/1BgfoS1_Co7DlKtMPYsJ-98vp4ctrVaPL?authuser=1

Reply:
Let's make the casts explicit using [`tf.cast`](https://www.tensorflow.org/api_docs/python/tf/cast)

Reply:
casted zero explicitly. Please have a look.